### PR TITLE
Create `environment.yml` file.

### LINKED ISSUE #350: Setup environment.yml for creating a development Conda environment
Reported by @czgdp1807 at https://github.com/lcompilers/lpython/pull/234#issuecomment-1093855868:

IMO, we can also an environment.yml file separately under the project root. Reason being, it helps new devs to quickly setup by just executing, conda create env -f environment.yml, instead of going through a bunch links on the web to figure how to do it all. In any case having such a file would be really helpful.

### PR DESCRIPTION
**Pull request by akshanshbhatt**
Closes #350

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,10 +30,9 @@ https://github.com/conda-forge/miniforge/#download
 
 ## Compile LPython
 
-Create a Conda environment:
+Create a Conda environment using the preexisting environment.yml file:
 
-    conda create -c conda-forge -n lp llvmdev=11.0.1 bison=3.4 re2c python cmake make toml numpy
-    conda activate lp
```
Comment (certik) [inline README.md:36]: You still need to activate the environment, so leave the `conda activate lp` line here.

**Comment by certik** [comment]
I tested this, and it works for me (on Apple M1). So I think this is good. I left a comment above, otherwise this is good to go in. Thanks!

File: README.md (hunk 1 above)
Comment (akshanshbhatt) [reply README.md:36]: Removed it by mistake. Thanks.

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  - conda-forge
+dependencies:
+  - bison=3.4
+  - cmake
+  - flake8
+  - llvmdev=11.0.1
+  - make
+  - numpy
+  - pip
+  - python=3.10.2
```
Comment (czgdp1807) [inline environment.yml:12]: I don't think its a good idea to write a specific python version here. `conda` should be able to figure out on its own what version of python should be installed on a specific system.

Comment (Smit-create) [reply environment.yml:12]: I don't think we need python in `environment.yml`

Comment (czgdp1807) [reply environment.yml:12]: I see. So, how will someone update the reference tests i.e., `./run_tests.py -u`? By using system Python? I don't usually prefer using system Python when I am already creating a conda environment. Not sure about others.

Comment (Smit-create) [reply environment.yml:12]: I think `conda` automatically adds `python` while creating an environment.

**Pull request merged**